src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -256,6 +263,34 @@ private static ExecutorService newExecutorService() {
             }
         }
 
+        @Override
+        public void onFailure(StepContext context, Throwable t) {
+            super.onFailure(context, t);
+            if (t instanceof FlowInterruptedException) {
```

Comment:
Might be worth adding a system property to revert to the old behavior in case it causes problems for some users. For example, something like this:
```suggestion
            if (t instanceof FlowInterruptedException && !Boolean.getBoolean(TimeoutStepExecution.class.getName() + ".forceInterruption")) {
```

If you do this, I would mention the property in the release notes.

Reply:
Done, modulo the release note update, which I assume Carroll would take care of as the one doing the release.